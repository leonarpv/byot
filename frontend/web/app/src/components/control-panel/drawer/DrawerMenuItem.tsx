import * as React from 'react';
import {ListItem, ListItemIcon, ListItemText, makeStyles, Theme} from '@material-ui/core';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';
import {Link} from 'react-router-dom';

interface Props extends WithStyles<typeof styles> {
  icon?: React.ReactNode;
  children?: React.ReactNode;
  link: string;
  selected?: boolean;
}

const styles = (theme: Theme) => ({
  root: {
    position: 'relative' as 'relative',
  },
  highlight: (props: Props) => ({
    width: theme.spacing(1 / 2),
    backgroundColor: props.selected ? theme.palette.grey[300] : 'transparent',
    transition: `background-color ${theme.transitions.duration.standard}ms`,
    display: 'block',
    left: 0,
    top: 0,
    height: '100%',
    position: 'absolute' as 'absolute',
  }),
  icon: (props: Props) => ({
    color: props.selected ? theme.palette.secondary.main : theme.palette.grey[500],
    transition: `color ${theme.transitions.duration.standard}ms`,
  }),
  textPrimary: (props: Props) => ({
    color: props.selected ? theme.palette.grey[900] : theme.palette.grey[500],
    transition: `color ${theme.transitions.duration.standard}ms`,
    fontWeight: 500,
  }),
});
const useStyles = makeStyles(styles);

export const DrawerMenuItem: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  return (
    <ListItem button component={Link} to={props.link} classes={{root: styles.root}}>
      <span className={styles.highlight} />
      <ListItemIcon classes={{root: styles.icon}}>{props.icon}</ListItemIcon>
      <ListItemText classes={{primary: styles.textPrimary}} primary={props.children} />
    </ListItem>
  );
};
