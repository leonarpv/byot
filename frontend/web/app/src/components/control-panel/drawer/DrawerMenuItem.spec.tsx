import * as React from 'react';
import {render} from '@testing-library/react';
import {DrawerMenuItem} from './DrawerMenuItem';

describe('<DrawerMenuItem/>', () => {
  it('should render', () => {
    const {container} = render(<DrawerMenuItem link="link" />);
    expect(container).toMatchSnapshot();
  });
  it('should render selected', () => {
    const {container} = render(<DrawerMenuItem link="link" selected />);
    expect(container).toMatchSnapshot();
  });
});
