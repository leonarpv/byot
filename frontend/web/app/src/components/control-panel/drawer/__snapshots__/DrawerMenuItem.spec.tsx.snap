// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerMenuItem/> should render 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-root-1 MuiListItem-gutters MuiListItem-button"
    data-testid="control-panel-drawer-menu-item"
    role="button"
    tabindex="0"
    to="link"
  >
    <span
      class="makeStyles-highlight-2 makeStyles-highlight-5"
    />
    <div
      class="MuiListItemIcon-root makeStyles-icon-3 makeStyles-icon-6"
    />
    <div
      class="MuiListItemText-root"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;

exports[`<DrawerMenuItem/> should render selected 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-root-8 MuiListItem-gutters MuiListItem-button"
    data-testid="control-panel-drawer-menu-item"
    role="button"
    tabindex="0"
    to="link"
  >
    <span
      class="makeStyles-highlight-9 makeStyles-highlight-12"
    />
    <div
      class="MuiListItemIcon-root makeStyles-icon-10 makeStyles-icon-13"
    />
    <div
      class="MuiListItemText-root"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;
